pkg/controller.v1/common/pod.go

```diff
@@ -86,7 +86,7 @@ func (jc *JobController) AddPod(obj interface{}) {
 		logger := commonutil.LoggerForPod(pod, jc.Controller.GetAPIGroupVersionKind().Kind)
 
 		if job == nil {
-			if pod.Labels[apiv1.GroupNameLabel] == jc.Controller.GetGroupNameLabelValue() {
+			if pod.Labels[apiv1.GroupNameLabel] != jc.Controller.GetGroupNameLabelValue() {
```

Comment:
good catch up

Reply:
@xfate123 

Please check logics here. https://github.com/kubeflow/common/blob/ac0c6e151f871bd7ed5673648b021f8351b5c663/pkg/controller.v1/common/job_controller.go#L321-L336

If we can not find the job, we will end reconcile loop and directly `return`. The only thing matters is if we want to have some meaningful log. 

There're several reason `job == nil` here

1. Kind doesn't match
2. GroupNameLabel doesn't match
3. Job doesn't exist. In this case, the pod is an orphan pod.

the 3rd one is the only case we want to persist the log. Does it make sense?

Reply:
@Jeffwan 
Thank you so much Jeff. Much clearer after your patient explanation.
But I am still a little confused
I think the three cases should be:
1.Kind doesn't match
2.Name unmatched
3. Name matched but UID unmatched
My understanding is that third case cannot prove this pod is an orphan pod. It can only prove that the controllerRef point s to a different job.
Thank you again. Really appreciate your help
Thank you

Reply:
You are right. I didn't follow logic in `resolveControllerRef`. UID mismatch is one of the cases.  We check `GroupNameLabel` again in caller side which I think is unnecessary. We use fixed value `kubeflow.org` for most of the operators.  

> My understanding is that third case cannot prove this pod is an orphan pod. It can only prove that the controllerRef point s to a different job.

I may not explain this clearly. you are right.  It could be a pod without Job ref or a pod point to different job (UID mismatch).

Reply:
@Jeffwan  Thank you so much for your clear explanation